clients/python/src/model_registry/utils.py

```diff
@@ -374,30 +375,48 @@ def save_to_oci_registry(  # noqa: C901 ( complex args >8 )
         dest_dir_cleanup = True
     local_image_path = Path(dest_dir)
 
-    # Set params
     params = {}
-
-    # User/pass
-    if auth:
-        usr_pass = auth.split(":")
-        params["username"] = usr_pass[0]
-        params["password"] = usr_pass[-1]
-    elif oci_username and oci_password:
-        params["username"] = oci_username
-        params["password"] = oci_password
-
-    backend_def.pull(base_image, local_image_path, **params)
-    # Extract the absolute path from the files found in the path
-    files = [file[0] for file in _get_files_from_path(model_files_path)]
-    oci_layers_on_top(local_image_path, files, modelcard)
-    backend_def.push(local_image_path, oci_ref, **params)
+    with temp_auth_file(auth) as auth_file:
+        if auth_file is not None:
+            params["authfile"] = auth_file.name
+        backend_def.pull(base_image, local_image_path, **params)
+        # Extract the absolute path from the files found in the path
+        files = [file[0] for file in _get_files_from_path(model_files_path)]
+        oci_layers_on_top(local_image_path, files, modelcard)
+        backend_def.push(local_image_path, oci_ref, **params)
 
     # Return the OCI URI
     if dest_dir_cleanup:
         shutil.rmtree(dest_dir)
     return f"oci://{oci_ref}"
 
 
+@overload
+def temp_auth_file(auth: str) -> AbstractContextManager[TextIO]:
+    ...
+
+
+@overload
+def temp_auth_file(auth: None) -> AbstractContextManager[None]:
+    ...
+
+
+@contextmanager
+def temp_auth_file(auth: str | None) -> AbstractContextManager[TextIO | None]:
+    """Create a temporary auth file with optional auth data.
+
+    If auth is None, yields None. Otherwise creates a temporary JSON file
+    containing the auth string and yields the file handle.
+    """
+    if auth is None:
+        yield None
+    else:
+        with tempfile.NamedTemporaryFile(mode="w+", encoding="utf-8", suffix=".json") as temp_auth_file:
+            temp_auth_file.write(auth)
+            temp_auth_file.flush()
+            yield temp_auth_file
```

Comment:
Looking at this again, I'm realizing it's probably best to close the file and use `delete=False` and handle the file cleanup ourselves.

It should work as is on posix/unix, but this sort of thing would **not** work on windows I believe, and closing files provides an even better guarantee that nothing is buffered than just calling `flush`.